src/useLatest.ts

```diff
@@ -0,0 +1,13 @@
+import { useRef, useEffect, MutableRefObject } from 'react';
+
+const useLatest = <T>(value: T): MutableRefObject<T> => {
+  const latest = useRef<T>(value);
+
+  useEffect(() => {
+    latest.current = value;
+  }, [value]);
+
+  return latest;
+};
```

Comment:
Do we need to put this into `useEffect`? Should it be out of the `useEffect`, like so:

```suggestion
const useLatest = <T>(value: T): MutableRefObject<T> => {
  const latest = useRef<T>(value);
  latest.current = value;
  return latest;
};
```

My guess is it should still work and be more performant, and new value updates will happen instantly instead of when scheduled by `useEffect`.

Reply:
Indeed it works, I did the change in 5488f5e.

My mileage may vary but I thought that it was a bit of an anti-pattern to perform side-effects at render in React?

Reply:
@ClementParis016 @streamich I understand this was merged long ago, but there is little response in the issue tracker at the moment. 
Refs should not be mutated outside of useEffect or an event handler as they will not be compatible with concurrent mode.

Reply:
@chrisk-7777 as concurrent mode is still experimental for a while i dont believe it will release any time soon - there is no sense to try adjust to it, as literally everything can change.

Reply:
Even if you ignore concurrent mode, its still generally recommended to avoid by the Hooks FAQ: 
https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables

> Unless you’re doing lazy initialization, avoid setting refs during rendering — this can lead to surprising behavior. Instead, typically you want to modify refs in event handlers and effects

But up to you guys I guess 🤷